### PR TITLE
Optimize Dockerfile.native for smaller docker-image size

### PR DESCRIPTION
**Pull request by haraldatbmw**
Optimizing the native Dockerfile by combining the **copy** with the **chown** command. 
With that in place one docker-layer can be dropped. 
This safes 1x the size of the native executable.

For background read this blog-post: https://blog.mornati.net/docker-images-and-chown/

Tested the changed  Dockerfile.native with Windows10 and Ubuntu Linux, both using Docker 19.03.8.

**before change:**
```
docker images dev/simple:1.0
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
dev/simple          1.0                 db1611fb072c        28 seconds ago      164MB
```
```
docker history dev/simple:1.0
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
db1611fb072c        7 seconds ago       /bin/sh -c #(nop)  CMD ["./application" "-Dq…   0B                  
f642cf578ba7        7 seconds ago       /bin/sh -c #(nop)  USER 1001                    0B                  
aa93812566e5        8 seconds ago       /bin/sh -c #(nop)  EXPOSE 8080                  0B                  
ecd06c4fdc15        8 seconds ago       /bin/sh -c chmod 775 /work /work/application…   29.1MB              
00d68ba8db63        10 seconds ago      /bin/sh -c #(nop) COPY file:7f4154ec775be2da…   29.1MB              
7381d031be10        6 weeks ago         /bin/sh -c #(nop) WORKDIR /work/                0B                  
6ce38bb5210c        2 months ago                                                        4.23kB              
<missing>           2 months ago                                                        106MB               Imported from -
```

**after change:**
```
> docker images dev/simple:1.0
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
dev/simple          1.0                 97a4faa5c9bd        35 minutes ago      135MB
```
```
> docker history dev/simple:1.0
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
97a4faa5c9bd        32 minutes ago      /bin/sh -c #(nop)  CMD ["./application" "-Dq…   0B                  
0f8a16d0616d        32 minutes ago      /bin/sh -c #(nop)  USER 1001                    0B                  
45c1bf0e7f72        32 minutes ago      /bin/sh -c #(nop)  EXPOSE 8080                  0B                  
6c677db83e3a        32 minutes ago      /bin/sh -c #(nop) COPY --chown=1001:rootfile…   29.1MB              
7381d031be10        6 weeks ago         /bin/sh -c #(nop) WORKDIR /work/                0B                  
6ce38bb5210c        2 months ago                                                        4.23kB              
<missing>           2 months ago                                                        106MB               Imported from -
```

### CODE REVIEW

File: devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl [hunk 1]
```diff
@@ -16,13 +16,7 @@
...
 FROM registry.access.redhat.com/ubi8/ubi-minimal:8.1
 WORKDIR /work/
-COPY ${build_dir}/*-runner /work/application
-
-# set up permissions for user `1001`
-RUN chmod 775 /work /work/application \
-  && chown -R 1001 /work \
-  && chmod -R "g+rwX" /work \
-  && chown -R 1001:root /work
+COPY --chown=1001:root ${build_dir}/*-runner /work/application
```
Comment (gsmet) [inline devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: just a quick comment: we lost the `chmod` in passing. It's not a problem?

/cc @cescoffier

Comment (haraldatbmw) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: @gsmet I see, but not a problem in my tests.

Comment (haraldatbmw) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: @gsmet Don't know if podman or other docker versions have problems.
Also my Windows docker deamon does it correct.

> SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.

Comment (gsmet) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: Well, you're the one who insisted on it :)

https://github.com/quarkusio/quarkus/pull/6461

Comment (haraldatbmw) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: Yes, seams my newer docker daemon fixed this already.

Comment (geoand) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: @cescoffier IIRC you added these commands, mind taking a look?

Comment (cescoffier) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: These commands were the recommendation made by Ope3nshift and Podman. 

The change looks good but it needs to be tested carefully.

**Comment by haraldatbmw** [comment]
Don't know why the Quarkus CI / Native Test - Security2 fails...

File: devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl (hunk 1 above)
Comment (haraldatbmw) [reply devtools/platform-descriptor-json/src/main/resources/templates/dockerfile-native.ftl:19]: Do you have automated tests for deployment into Openshift and with Podman?

**Comment by geoand** [comment]
> Don't know why the Quarkus CI / Native Test - Security2 fails...

Flaky test unfortunately

**Comment by haraldatbmw** [comment]
> > Don't know why the Quarkus CI / Native Test - Security2 fails...
> 
> Flaky test unfortunately

So do I have to care about it - retrigger the CI stuff?

**Comment by geoand** [comment]
Nah, that's not necessary since your change in no way affects that test.

**Comment by cescoffier** [comment]
I’ve going to test with podman tomorrow.

**Comment by cescoffier** [comment]
Just tried with podman, it works!

**Comment by gsmet** [comment]
Ok, let's try it and we will see if some users complain.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks @haraldatbmw !

**Pull request merged**